### DISCUSSION TITLE
How to get rid of false positive motion events inside Surveillance Station

### DISCUSSION THREAD

**Discussion by Jazor25** [djdd87/SynoAI#192] [Q&A]
Hi, this is a great project! Probably very helpful because my camera's motion detection at the front door produces lots of false positives.

I use 24/7 recording for my camera in Surveillance Station and use the web portal itself or the DS Cam app to take a look at recent recordings of detected motion events.

How I understand the described workflow, you use the cameras Motion detection as a trigger and validate via SynoAI if the detected motion was valid and then send a notification via some external service like Telegram or via PUSH to Home Assistant.
Is it also possible (and if it is, how do I do it) to improve Synologys own history of detected motion events (get rid of false positives)? If the front door camera detects a motion and SynoAI identifies this event as a false positive, is it possible to discard this motion event on the Synology itself? I would hope to get rid of false positives when I take a look at recent motion events on the DS Cam app.

Hope this makes sense and somebody can let me know if and how this is possible.
Thank you!

**Comment by djdd87** [comment]
Hey, unfortunately I don't see anything in the API to delete motion events, nor do I believe that SSS passes any sort of unique identifier on webhook calls that could be used to pass back. It's been a while since I dug into the API docs though and I'm struggling to find time to work on this project at the moment due to the sheer volume of other work I've got on.

**Comment by JoeyJoe6** [comment]
Here is my work around for "get rid of false positives". Yes, you will use an additional SS licenses for this
1. Create a second IP camera entry, example: "front-door-camera-alerts"
2. Set your preference for event detection for camera "front-door-camera-alerts" in SS (either by camera or by SS)
3. In your original IP camera "front-door-camera", disable event detection completely
4. Create a Action Rule;
- INFORMATION: "front-door-camera-alerts", Triggered, Interruptible
- EVENT: Event1, Camera, device: front-door-camera-alerts, Event: Motion detection
- ACTION: Webhook, URL: [your synoai IP]:port/Camera/front-door-camera-alerts
5. Create a Action Rule; This rule will be used for positive detection from synoai
- INFORMATION: "Mark-front-door-camera-alert", Triggered, Interruptible
- EVENT: Event1, webhook, you will be provided a webhook url at the bottom
- ACTION: Camera, Device: front-door-camera, Action: Trigger Motion Event
* this rule will be used to mark your timeline for camera "front-door-camera" that a positive event has occurred
6. In your synoai appsetting.json file:
ADD these lines into your "Notifiers" section
	{
    "Type": "Webhook",
    "URL": "http://[synologyIP]:5000/webapi/entry.cgi?api=SYNO.SurveillanceStation.Webhook&method=\"Incoming\"&version=1&token="insert the token value from webook action rule created with in these quotes",
    "Method": "GET",
	"SendImage": "false",
	"Cameras": [ "front-door-camera-alerts" ]
    },
** note the area =Incoming was changed from the original webhook. Had to add in the \ to each side for some reason for it to work
** This section will be executed if a true positive is found by Synoai from camera "front-door-camera-alerts", and send the webhook to syonlogy SS, which in turn will mark the timeline for camera "front-door-camera" for easier and more accurate alerts. 
7. If not already done (as noted in the synoai docs)
- Add this entry into the "Camera" section of the appsetting.json file
   "Cameras": [
   	{
      "Name": "front-door-camera-alerts",
      "Types": [ "Person", "cat", "dog", "boat", "car", "truck" ], 
      "Threshold": 47,
      "MinSizeX": 150,
      "MinSizeY": 150
    },
]
** or what ever Types you want to detect. or just remove the TYPES line all together to mark anything found
